analysis/task_analysis_volume/group_level_analysis_vol.ipynb

```diff
@@ -0,0 +1,1013 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "12f735f0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "task='nback'\n",
+    "sessions = ['baseline','1year']\n",
+    "groups = ['HC','MM']\n",
+    "MNI_template_path = '../templates/tpl-MNI152NLin6Asym_res-01_desc-brain_T1w.nii.gz'\n",
+    "MNI_template_used = nibabel.load(MNI_template_path)\n",
+    "\n",
+    "if task == 'mid':\n",
+    "    contrasts=['HiWin+LoWin-HiNoWin-LoNoWin','HiLoss+LoLoss-AvoidHiLoss-AvoidLoLoss']\n",
+    "\n",
+    "elif task == 'sst':\n",
+    "    contrasts=['SuccStop-Go','UnsuccStop-Go','SuccStop-UnsuccStop']\n",
+    "    \n",
+    "elif task == 'nback':\n",
+    "    contrasts=['twoback-zeroback']\n",
+    "\n",
+    "#options are age, sex, total_cudit, THC_freq_month at the moment (HC is 0 for total_cudit and THC_freq)\n",
+    "#covariates = ['sex','age','total_cudit', 'THC_freq_month']\n",
+    "covariates = []\n",
+    "\n",
+    "#subgroup = 'dipstick_THC'\n",
+    "subgroup = ''\n",
+    "\n",
+    "#set to true if MM participants with CUD at baseline should be excluded from the analysis\n",
+    "rm_CUD = True\n",
+    "\n",
+    "#set this to the ones we want to include with values for fd_mean and fd_perc as keys\n",
+    "desired_outliers = {'tsnr':[],'snr':[],'gsr_x':[],'gsr_y':[],'fd_mean':0.2,'fd_perc':0.3}\n",
+    "#desired_outliers = {}\n",
+    "\n",
+    "stat_max=[]\n",
+    "stat_min=[]\n",
+    "effect_max=[]\n",
+    "effect_min=[]\n",
+    "\n",
+    "\n",
+    "#loop through all sessions, groups, contrasts\n",
+    "for ses in sessions:\n",
+    "    for group in groups:        \n",
+    "        for contrast in contrasts:\n",
+    "            #get group average group level contrast as the output\n",
+    "            contrast_outputs, mask, part_count, participants = get_group_average_contrast(contrast,ses,group,task,covariates,subgroup,rm_CUD,desired_outliers)\n",
+    "            \n",
+    "            #account for HC not having 1year scans\n",
+    "            if contrast_outputs:\n",
+    "                \n",
+    "                print(f'For task contrast {contrast}, we will print the covariate-based contrasts in the following order: {contrast_outputs.keys()}:')\n",
+    "                \n",
+    "                for slope in contrast_outputs.keys():\n",
+    "                    \n",
+    "                    contrast_output = contrast_outputs[slope]\n",
+    "                    \n",
+    "                    #compute FDR-corrected z-score thresholdmap\n",
+    "                    #note that this does not have a cluster threshold\n",
+    "                    #this image is masked by the mask intersection across all subjects\n",
+    "                    stat_map = contrast_output['z_score']\n",
+    "                    effect_map = contrast_output['effect_size']\n",
+    "\n",
+    "                    #for \"positive\" activation difference (nicer visualization)\n",
```

Comment:
if this is used, make sure that the figure description says so. why not just use the reverse contrast definition in the first place?

Reply:
it was a quick fix to not have to rerun things. do you think it's worth editing it all?

Reply:
this is more a notion of the kind of thing that gets lost in details. I don't think you need to change the code at the moment, but make sure that in the paper wherever this contrast is mentioned or shown, the appropriate contrast is described.

Reply:
also make a note in the place where the contrast is specified in the script that the plots reverse the effect.